app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java

```diff
@@ -130,16 +140,19 @@ public void bind(int position) {
                     callback.showAlert(R.string.media_detail_description, R.string.description_info));
 
             } else {
+                removeButton.setVisibility(View.VISIBLE);
                 captionInputLayout.setEndIconDrawable(null);
                 descInputLayout.setEndIconDrawable(null);
             }
 
+            removeButton.setOnClickListener(v -> removeDescription(uploadMediaDetail, getAdapterPosition()));
+
             captionItemEditText.addTextChangedListener(new AbstractTextWatcher(
-                    captionText -> uploadMediaDetails.get(position).setCaptionText(captionText)));
+                    captionText -> uploadMediaDetails.get(getAdapterPosition()).setCaptionText(captionText)));
```

Comment:
these `getAdapterPosition`s should be replaced with position too

Reply:
This one is intentionally done! It was leading to an incorrect position which crashes the app.

Reply:
Now it also makes me wonder if I should revert it
```
if (position == 0) {
    eventListener.onPrimaryCaptionTextChange(value.length() != 0);
}
```

Reply:
I think that was the HLRV causing desyncs as I got those crashes with these in place

Reply:
Cool! worked!